Assets/MixedRealityToolkit/Utilities/MRTKServicesUtils.cs

```diff
@@ -0,0 +1,110 @@
+    /// </summary>
+    public static class MRTKServicesUtils
+    {
+        private static IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system
+        /// </summary>
+        public static IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        private static IMixedRealityBoundarySystem boundarySystem = null;
+
+        /// <summary>
+        /// The active instance of the boundary system
+        /// </summary>
+        public static IMixedRealityBoundarySystem BoundarySystem
+        {
+            get
+            {
+                if (boundarySystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityBoundarySystem>(out boundarySystem);
+                }
+                return boundarySystem;
+            }
+        }
+
+        private static IMixedRealitySpatialAwarenessSystem spatialAwarenessSystem = null;
+
+        /// <summary>
+        /// The active instance of the spatial awareness system
+        /// </summary>
+        public static IMixedRealitySpatialAwarenessSystem SpatialAwarenessSystem
+        {
+            get
+            {
+                if (spatialAwarenessSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealitySpatialAwarenessSystem>(out spatialAwarenessSystem);
+                }
+                return spatialAwarenessSystem;
+            }
+        }
+
+        private static IMixedRealityTeleportSystem teleportSystem = null;
+        /// <summary>
+        /// The active instance of the teleport system
+        /// </summary>
+        public static IMixedRealityTeleportSystem TeleportSystem
+        {
+            get
+            {
+                if (teleportSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityTeleportSystem>(out teleportSystem);
+                }
+
+                return teleportSystem;
+            }
+        }
+
+        private static IMixedRealityDiagnosticsSystem diagnosticsSystem = null;
+        /// <summary>
+        /// The active instance of the diagnostics system
+        /// </summary>
+        public static IMixedRealityDiagnosticsSystem DiagnosticsSystem
+        {
+            get
+            {
+                if (diagnosticsSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityDiagnosticsSystem>(out diagnosticsSystem);
+                }
+                return diagnosticsSystem;
+            }
+        }
+
+        public static void InvalidateReferences()
```

Comment:
consider adding a flags param to allow invalidating individual services (or all)

Reply:
I am not going to add a flag for now. I want to keep things simple and resetting all local references does not have much of an impact. They will just be lazy-loaded again

Reply:
I am going to keep it simple for now. It doesn't really impact much to just do another lazy load on next iteration after full reset